eng/Packages.Data.props

```diff
@@ -89,7 +89,7 @@
     <PackageReference Update="System.ClientModel" Version="1.0.0" />
     <PackageReference Update="System.IO.Hashing" Version="6.0.0" />
     <PackageReference Update="System.Memory" Version="4.5.4" />
-    <PackageReference Update="System.Memory.Data" Version="1.0.2" />
+    <PackageReference Update="System.Memory.Data" Version="6.0.0" />
```

Comment:
This one is not an STJ dependency but is also overdue for an upgrade so that we could take advantage of APIs in newer versions.